File: python/grains/grains.py
def on_square(integer_number):
    check_value(integer_number)
    return 1 << (integer_number - 1)


def total_after(integer_number):
    check_value(integer_number)
    return sum([on_square(number) for number in range(1, integer_number + 1)])

def check_value(integer_number):
    if integer_number not in range(1, 65):
        raise ValueError("the value must be beetween 1 and 64 inclusive")

File: csharp/grains/Grains.cs
public static class Grains
{
    private const int NumberMaxSquares = 64;
    private const int NumberMinSquare = 1;
    
    public static ulong Square(int n)
    {
        if (n < NumberMinSquare || n > NumberMaxSquares)
            throw new System.ArgumentOutOfRangeException("the value must be beetween 1 and 64 inclusive");
        return (ulong)1 << (n - 1);
    }

    public static ulong Total()
    {
        ulong total = 0;
        
        for (int i = NumberMinSquare; i <= NumberMaxSquares; i++)
        {
            total += Square(i);
        }
        return total;
    }
}